lib/child_process.js

```diff
@@ -622,13 +611,8 @@ function validateMaxBuffer(maxBuffer) {
 
 function validateKillSignal(killSignal) {
   if (typeof killSignal === 'string' || typeof killSignal === 'number') {
-    killSignal = lookupSignal(killSignal);
-
-    if (killSignal === 0)
```

Comment:
I think zero is a special case only in `ChildProcess#kill`. So I special cased it there.